spec/forms/state_file/notification_preferences_form_spec.rb

```diff
@@ -60,63 +60,92 @@
     end
 
     context "when email_notification_opt_in is present" do
-      let(:valid_params) do
-        {
-          email_notification_opt_in: "yes",
-          email_address: "test@example.com"
-        }
-      end
-
-      subject(:form) { described_class.new(intake, valid_params) }
-
-      it "is valid" do
-        expect(form).to be_valid
-      end
-
-      context "when email address is invalid" do
-        let(:invalid_params) do
+      context "with no existing email address" do
+        let(:valid_params) do
           {
             email_notification_opt_in: "yes",
-            email_address: "invalid-email"
+            email_address: "test@example.com"
           }
         end
-        subject(:form) { described_class.new(intake, invalid_params) }
 
-        it "is invalid" do
-          form.valid?
-          expect(form).not_to be_valid
+        subject(:form) { described_class.new(intake, valid_params) }
+
+        it "is valid" do
+          expect(form).to be_valid
         end
 
-        it "adds an error to the email_address attribute" do
-          form.valid?
-          expect(form.errors[:email_address]).to be_present
+        it "updates the intake with the provided attributes" do
+          form = described_class.new(intake, valid_params)
+          expect { form.save }.to change(intake, :email_notification_opt_in).from("unfilled").to("yes")
+                                                                            .and change(intake, :email_address).to("test@example.com")
+        end
+
+        context "when email address is invalid" do
+          let(:invalid_params) do
+            {
+              email_notification_opt_in: "yes",
+              email_address: "invalid-email"
+            }
+          end
+          subject(:form) { described_class.new(intake, invalid_params) }
+
+          it "is invalid" do
+            form.valid?
+            expect(form).not_to be_valid
+          end
+
+          it "adds an error to the email_address attribute" do
+            form.valid?
+            expect(form.errors[:email_address]).to be_present
+          end
+        end
```

Comment:
[dust] can we fold the "is invalid" and "adds an error to the email_address attribute" into one `it` block? I do see it as a pattern here but wondering if there's a good reason why these need to be separate...open to either, interested to hear what your thoughts are here